src/main/java/io/confluent/examples/streams/microservices/EmailService.java

```diff
@@ -40,7 +44,24 @@ public EmailService(final Emailer emailer) {
   public void start(final String bootstrapServers) {
     streams = processStreams(bootstrapServers, "/tmp/kafka-streams");
     streams.cleanUp(); //don't do this in prod as it clears your state stores
+    final CountDownLatch startLatch = new CountDownLatch(1);
+    streams.setStateListener((newState, oldState) -> {
+      if (newState == State.RUNNING && oldState == State.REBALANCING) {
+        startLatch.countDown();
+      }
+
+    });
     streams.start();
+    final boolean rebalanceCompleted;
```

Comment:
nit: is this variable necessary? It seems like :
```java
if (!startLatch.await(60, TimeUnit.SECONDS) {
        throw new RuntimeException("Streams never finished rebalancing on startup");
}
```

would do the same thing?

Reply:
ack